app/javascript/src/js/audio/ohms_search.js

```diff
@@ -0,0 +1,463 @@
+};
+
+// Scrolls browser to show domID passed in.
+//
+// While it would be lovely to just use built-in scrollIntoView, we need to take
+// care of a few things that doesn't:
+//
+//   * Avoid fixed navbar on top, we don't want to scroll so the thing we want
+//     to show is behind the navbar.
+//
+//   * If id is in a tab that isn't currently visible, switch to that tab.
+//
+//   * If id is in a bootstrap collapsed that isn't currently expanded, expand it.
+//     (intended for Table of Contents section)
+//
+// Second optional argument is either "smooth" or "auto", with "auto" being the
+// default. As in HTML5 scroll functions which it will be passed to, "smooth"
+// will do a visible smooth scroll, "auto" just jumps to position.
+Search.scrollToId = function(domID, scrollBehavior) {
+  if (scrollBehavior == undefined) {
+    scrollBehavior = "smooth";
+  }
+
+  // without block:center, it ends up scrolling under our fixed navbar, gah!
+  // this seems to be good enough.
+  var element = document.getElementById(domID);
+
+  // If it's in a tab, and the tab isn't currently shown, make it shown
+  var tabPane = $(element).closest(".tab-pane");
+  if (tabPane) {
+    // annoyingly, have to get the actual tab link that corresponds to
+    // the pane
+    var tab = $(".nav-link[href='#" + tabPane.attr("id") + "']");
+    tab.tab("show");
+  }
+
+  // If our target element CONTAINS a bootstrap collapsible that is collapsed,
+  // show it. This is intended for our ToC accordion.
+  var collapsible = $(element).find(".collapse");
+  if (collapsible && ! collapsible.hasClass("show")) {
+    collapsible.collapse("show");
+  }
+
+  var elTop = $(element).offset().top;
+  var navbarHeight = $("#ohmsAudioNavbar").height();
+
+  window.scrollTo({top: elTop - navbarHeight, behavior: scrollBehavior});
+};
+
+// Execute a search, in response to a search submit. Find transcript and index
+// results, set search results state, draw search results on screen.
+Search.onSearchSubmit = function(event) {
+  event.preventDefault();
+
+  Search.clearSearchResults();
+
+  var query = $(event.target).find("*[data-ohms-input-query]").val();
+
+  if (query == "") {
+    return;
+  }
+
+  var transcriptResults = Search.searchTranscript(query);
+  Search.currentTranscriptResults = new Search.SearchResults(
+    $("*[data-ohms-search-results]").get(0),
+    transcriptResults,
+    "transcript"
+  );
+  $("*[data-ohms-hitcount='transcript']").html('<span class="badge badge-pill badge-danger">' + transcriptResults.length + '</span>');
+
+  var indexResults = Search.searchIndex(query);
+  Search.currentIndexResults = new Search.SearchResults(
+    $("*[data-ohms-search-results]").get(0),
+    indexResults,
+    "index"
+  );
+  $("*[data-ohms-hitcount='index']").html('<span class="badge badge-pill badge-danger">' + indexResults.length + '</span>');
+
+  Search.currentResults().draw();
+  Search.currentResults().scrollToCurrentResult();
```

Comment:
The scrollToCurrentResult line should be wrapped in an `if(Search.currentResults().results.length) {`

Reply:
Thanks! Did you reproduce an error condition without this, verified something going wrong?

Reply:
Yeah - just if there are zero results there's an error somewhere in scrollToCurrentResult. You could also return at the beginning of scrollToCurrentResult - whatever works.

Reply:
awesome, thanks for catching this, I will try to reproduce before fixing, just so I understand what I'm fixing.

Reply:
Reproduced!  

Normally it defaults to starting with "Table of Contents". If there are no hits in ToC, but htere is a hit in Transcript.... should it switch you to Transcript instead to show you the hit? Or just keep showing you no hits in ToC?  

Not sure, not sure it matters or we'll pick the right UX anyway, so I guess I'll just do the simplest thing, which is probably show you the no hits in ToC.